packages/data/lib/src/database/dao/people_dao.dart

```diff
@@ -0,0 +1,14 @@
+import 'package:data/src/database/dto/people_with_image_dto.dart';
+import 'package:floor/floor.dart';
+
+@dao
+abstract class PeopleDao {
+  @Insert(onConflict: OnConflictStrategy.replace)
+  Future<List<int>> insertPeoples(List<PeopleWithImageDto> peoples);
+
+  @Query('SELECT * FROM PeopleWithImageDto WHERE movieId = :movieId')
+  Future<List<PeopleWithImageDto>> fetchPeoplesByMovieId(int movieId);
+
+  @Query('DELETE FROM PeopleWithImageDto WHERE movieId NOT IN (:idList)')
+  Future<void> deleteIfIdIsNotInList(List<int> idList);
```

Comment:
u shouldn't have separate query for deleting record from PeopleWithImageDto. if PeopleWithImageDto will have the foreign key to primary key from Movie table, deleting record from Movie should delete records from people by cascade